### PR TITLE
Oshan uses a transception array instead of shipping launchers 

### PR DESCRIPTION
**Pull request by pgmzeta**
[station systems][feature]

## About the PR 
Replaces Oshan's cargo intake/outake with transception array, pads, and computers.

Supporting work:
* Add a custom meteorhit effect to the transception array (yeets battery, requires full repair)
* Add repair steps in the transception array's help message
* Add a new define `CARGO_TRANSPCETION`
    * Applied on maps using transception arrays (Nadir, Oshan)
    * Replaces relevant `MAP_OVERRIDE_NADIR` ifdefs
* Slight wording tweak to the modified transception pocket guide

## Why's this needed? 
Active, 'charged up' hotspots along any portion of the cargo lane means crates disappear into the trench without warning, and often without intent. It's difficult to get the crates back, and repairing the lanes is dangerous with the large number of automatic crate recieves (mail box, special delivery, etc). Crates being launched through water is physics-defying, too :^)

## Screenshots
cargo backroom & array
image

fullbright / zoomed out
image

front office changes include a transception pad and console as well
image

## Changelog 

```changelog
(u)glowbold
(*)Oshan utilizes the latest in transception array technology to facilitate cargo needs.
```

### CODE REVIEW

**Comment by TobleroneSwordfish** [comment]
Pinging @Kubius for thoughts because they made this specifically for Nadir originally

**Review by Kubius — COMMENTED** [review]
No big objection here. The array was originally considered a prototype, but there's no reason the tech can't make its way to other suitable locations once it shakes out.

The main thing I'd be mindful of is the site arrangement for the array. The console that reports array status has some control over its function, so it should be somewhere at least mildly secure, and it should have a sizable power cell in the area containing the array itself, since transception requires a fair amount of power.

File: _std/mapDefines.dm [hunk 1]
```diff
@@ -46,9 +46,11 @@
 #define UNDERWATER_MAP 1
 #define SCIENCE_PATHO_MAP 1
 #define HOTSPOTS_ENABLED 1
+#define CARGO_TRANSCEPTION 1
 
 #elif defined(MAP_OVERRIDE_NADIR)
 #define UNDERWATER_MAP 1
+#define CARGO_TRANSCEPTION 1
```
[the remark this answers is not in the document]
Comment (pgmzeta) [reply _std/mapDefines.dm:53]: up on line 49

**Comment by pgmzeta** [comment]
> No big objection here. The array was originally considered a prototype, but there's no reason the tech can't make its way to other suitable locations once it shakes out.
> 
> The main thing I'd be mindful of is the site arrangement for the array. The console that reports array status has some control over its function, so it should be somewhere at least mildly secure, and it should have a sizable power cell in the area containing the array itself, since transception requires a fair amount of power.

I matched the 5000 unit power cell for the area APC as with Nadir's transception APC, and the console is in the cargo-access corridor behind cargo. It's a bit exposed by being on the outside, but the windows in the backroom keep it in sight. If sea-meteors cause a consistent 1 hit KO, there may be some more involved shuffling required.

**Comment by Kubius** [comment]
One minor supplementary detail is that there's stuff generating on top of the array in the preview and that looks a little bit wonky. Isn't gamebreaking, but could potentially do with adjustment. No-generate fluid turf is one option, the other is tinkering with how the stuff is layered (I'd probably go for the former).

**Comment by pgmzeta** [comment]
added nospawn for the transception turfs in the transception array bowl.

also added a custom meteorhit that pops the battery out and requires full repairs, so it getting hit by a meteor is serious but it doesn't qdel.

File: code/WorkInProgress/nadir_antenna.dm [hunk 2]
```diff
@@ -298,6 +298,38 @@ TYPEINFO(/obj/machinery/communications_dish/transception)
...
+	src.intcap_door_open = TRUE
+	src.repair_status = ARRAY_INTEG_WELD0
+	if (src.intcap)
+		var/obj/item/cell/former_cell = src.intcap
+		former_cell.set_loc(get_turf(src))
+		var/turf/target = get_offset_target_turf(src.loc, (rand(5)-rand(5)), (rand(5)-rand(5)))
+		former_cell.throw_at(target,5,1)
+		src.intcap = null
+	src.UpdateIcon()
+	src.visible_message(SPAN_ALERT("<b>[src]'s internal capacitor is violently ejected!</b>"))
```
Comment (Kubius) [inline code/WorkInProgress/nadir_antenna.dm:331]: If I'm reading this correctly, the ejection message will play even if the cell is already ejected. Could have this in the if(src.intcap) block and then have an alternate "is severely damaged" message in an else.

**Review by flrsh — COMMENTED** [review]
Apologies for the late review. I am not a fan of transcription arrays and strongly prefer the current shipping system. I have not had the time to look at the code, but I wonder if there's a way to either make the crate path specifically immune to opening up holes to trench or to make objects moving above a certain speed not fall into the trench.

**Comment by Kubius** [comment]
Should be entirely possible. Regular pitfalls have a 0.3 second delay before things will fall down them, but it tries the fall wherever the crate ends up being; if the fall check instead retried on the *same* tile after the delay when the falling object is a sellable (crate or artifact), this would mean the object wouldn't fall if it had moved to a different pitfall since that delay.

**Comment by AXLFAN2000** [comment]
Worth noting the other problem this addresses (that I see happen in round a lot more often than the hotspot/trench situation) is crates hitting fish and getting backed up that way.  If this gets denied, some way to prevent fish from stopping cargo crates would be nice too

**Pull request closed without merging**